File: CS/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.UserDesigner;
// ...

namespace docNewReportWizard {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void simpleButton1_Click(object sender, EventArgs e) {
            string fileName = CreateReport();

            this.BeginInvoke(new MethodInvoker(
                () => {
                    PreviewReport(fileName);
                }
                ));
        }

        private string CreateReport() {
            string fileName = "temp.repx";
            XRDesignForm form = new XRDesignForm();
            form.DesignMdiController.CreateNewReportWizard();
            form.ActiveDesignPanel.BeginInvoke(new MethodInvoker(() => { form.ActiveDesignPanel.Report.SaveLayout(fileName); }));
            return fileName;
        }
        private void PreviewReport(string fileName) {
            XtraReport newReport = XtraReport.FromFile(fileName, true);
            newReport.ShowPreview();
        }

    }
}

File: VB/Form1.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.UserDesigner
' ...

Namespace docNewReportWizard
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
            Dim fileName As String = CreateReport()

            Me.BeginInvoke(New MethodInvoker(Sub() PreviewReport(fileName)))
        End Sub

        Private Function CreateReport() As String
            Dim fileName As String = "temp.repx"
            Dim form As New XRDesignForm()
            form.DesignMdiController.CreateNewReportWizard()
            form.ActiveDesignPanel.BeginInvoke(New MethodInvoker(Sub() form.ActiveDesignPanel.Report.SaveLayout(fileName)))
            Return fileName
        End Function
        Private Sub PreviewReport(ByVal fileName As String)
            Dim newReport As XtraReport = XtraReport.FromFile(fileName, True)
            newReport.ShowPreview()
        End Sub

    End Class
End Namespace

